applications/tari_console_wallet/src/automation/commands.rs

```diff
@@ -159,19 +154,9 @@ pub async fn make_it_rain(
     );
     delay_for(Duration::from_millis(delay_ms)).await;
 
-    let num_txs = (txps * duration as f64) as usize;
-
     let mut tx_ids = Vec::new();
-    let started_at = Utc::now();
-
-    for i in 0..num_txs {
-        // Manage Tx rate
-        let actual_ms = (Utc::now() - started_at).num_milliseconds() as u64;
-        let target_ms = (i as f64 / (txps / 1000.0)) as u64;
-        if target_ms - actual_ms > 0 {
-            // Maximum delay between Txs set to 120 s
-            delay_for(Duration::from_millis((target_ms - actual_ms).min(120_000u64))).await;
-        }
```

Comment:
@delta1 and @SWvheerden, I think this is the problem right here. In the original implementation for the base node wallet, this calculation was specifically done with `i64` and not `u64` to prevent wrap around.

Reply:
The original implementation worked perfectly.

Reply:
Definitely the cause of the delay, but when subtracting u64s don't wrap around, it panics with: 

`thread 'main' panicked at 'attempt to subtract with overflow', applications/tari_console_wallet/src/automation/commands.rs:263:41` 

After changing the minus to a saturating sub, I ran a test successfully at 10 tx/sec for 5 seconds. But SW was running for 50 seconds and said that it immediately starting delaying by 2 minutes. 

Regardless, the logic needs a re-work if it's just about managing the maximum rate.